fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/common/MappedPoolTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * MappedPoolTests.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.common;
+
+import org.junit.jupiter.api.Test;
+import javax.crypto.Cipher;
+import java.security.GeneralSecurityException;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/** Tests of the MappedPool.
+ *
+ */
+public class MappedPoolTest {
+
+    public static String CIPHER = "DES/ECB/PKCS5Padding";
+    public static MappedPool<String, Cipher, GeneralSecurityException> MAPPED_POOL = new MappedPool<>(Cipher::getInstance);
+
+    @Test
+    public void testCipherPool() throws Exception {
+        Cipher lastCipher = null;
+        for (int i = 0; i < 100; i++) {
+            Cipher cipher = MAPPED_POOL.poll(CIPHER);
+            if (lastCipher != null) {
+                assertTrue(cipher.equals(lastCipher));
+                lastCipher = cipher;
+            }
+            assertNotNull(cipher);
+            assertTrue(MAPPED_POOL.offer(CIPHER, cipher));
+        }
+        assertTrue(MAPPED_POOL.getPoolSize(CIPHER) == 1);
```

Comment:
```suggestion
        assertEquals(1, MAPPED_POOL.getPoolSize(CIPHER));
```

(This will also need to import `assertEquals`, so this change itself won't compile.)